commons/Common.ml

```diff
@@ -617,15 +623,8 @@ let finalize f cleanup =
     cleanup ();
     res
   end
-  else begin
-    try
-      let res = f () in
-      cleanup ();
-      res
-    with e ->
-      cleanup ();
-      raise e
-  end
+  else
+    Fun.protect f ~finally:cleanup
```

Comment:
And Fun.protect does the right thing regarding exception? it reraises correctly?

Reply:
Yes! I checked. It does exactly what we would do, no extra magic: https://github.com/ocaml/ocaml/blob/trunk/stdlib/fun.ml#L30-L31